tests/imodels-access-backend-tests/src/BackendiModelsAccess.test.ts

```diff
@@ -181,6 +183,45 @@ describe("BackendIModelsAccess", () => {
     });
   });
 
+  describe("CreateNewIModel", () => {
+    it("copyAndPrepareBaselineFile should perform a wal checkpoint", async () => {
+      // cspell:disable-next-line
+      const filePath = path.join(testDownloadPath, "createnewimodel.bim");
+      const walPath = `${filePath}-wal`;
+      const loggerSpy = sinon.spy(Logger, "logWarning").withArgs("BackendIModelsAccess", "Wal file found while uploading file, performing checkpoint.", sinon.match.any);
+      await IModelHost.startup();
+      const testIModel = StandaloneDb.createEmpty(filePath, { rootSubject: { name: "test1"}, allowEdit: JSON.stringify({ txns: true })});
+      PhysicalModel.insert(testIModel, IModel.rootSubjectId, "TestModel");
+
+      expect(testIModel.isOpen).to.be.true;
+      expect(IModelJsFs.existsSync(walPath)).to.be.true;
+      expect(IModelJsFs.lstatSync(walPath)?.size).to.be.greaterThan(0);
+      const arg: CreateNewIModelProps = {
+        version0: filePath,
+        iModelName: "testimodel",
+        iTwinId: Guid.createValue()
+      };
+      // Expect database locked because we didn't save changes to end our transaction.
+      expect(() => (backendIModelsAccess as any).copyAndPrepareBaselineFile(arg)).to.throw("database is locked");
```

Comment:
couldn't we test the public `BackendHubAccess.createNewIModel` method here?

Reply:
We could, I guess my env file is out of date though because I was getting a 401 when trying to get the auth token. I changed the test to use the private function since I could actually iterate on that locally at the time, but can switch it back.

Reply:
switched to createnewimodel